Core/Src/main.c

```diff
@@ -87,16 +89,56 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  // The data to transmit.
+  // The 0b1 indicates a start bit.
+  // The 0b10000000 selects CH0 of the ADC, single-ended.
+  // The 0b0 is really a don't care for the "sample and hold" period.
+  uint8_t txBuffer[3] = {0b1, 0b10000000, 0b0};
+
+  // The buffer to receive the data.
+  uint8_t rxBuffer[3] = {0};
+
+  // The largest value the ADC can output (2^10 - 1).
+  const uint16_t maxAdcVal = 1023;
+
+  // Min and max PWM percentages for duty cycle.
+  const float minPWM = 0.05;
+  const float maxPWM = 0.10;
```

Comment:
it is very common practice to use ALL_CAPS notation for const variables… please do this for all consts.

also… great use of consts :) but I think txBuffer can also be const too since it will and should never change right?

Reply:
Done.